core/src/main/java/org/mskcc/cbio/oncokb/util/IndicatorUtils.java

```diff
@@ -454,10 +437,95 @@ private static Boolean isVUS(Alteration alteration) {
     }
 
     private static List<Alteration> findRelevantAlts(Gene gene, String alteration) {
+        Set<Alteration> relevantAlts = new LinkedHashSet<>();
         Alteration alt = AlterationUtils.getAlteration(gene.getHugoSymbol(), alteration,
             null, null, null, null);
         AlterationUtils.annotateAlteration(alt, alt.getAlteration());
-        return AlterationUtils.getRelevantAlterations(alt);
+
+        relevantAlts.addAll(AlterationUtils.getRelevantAlterations(alt));
+
+        Alteration revertAlt = AlterationUtils.getRevertFusions(alt);
+        if (revertAlt != null) {
+            relevantAlts.addAll(AlterationUtils.getRelevantAlterations(revertAlt));
+        }
+        return new ArrayList<>(relevantAlts);
+    }
+
+    public static Map<String, Object> findFusionGeneAndRelevantAlts(Query query) {
+        List<String> geneStrsList = Arrays.asList(query.getHugoSymbol().split("-"));
+        Set<String> geneStrsSet = new HashSet<>();
+        Gene gene = null;
+        List<Alteration> fusionPair = new ArrayList<>();
+        List<Alteration> relevantAlterations = new ArrayList<>();
+        Map<String, Object> map = new HashedMap();
+
+        if (geneStrsList != null) {
+            geneStrsSet = new HashSet<>(geneStrsList);
+        }
+
+        // Deal with two different genes fusion event.
+        if (geneStrsSet.size() >= 2) {
```

Comment:
would there be cases that the size is more than 2? If so, may I have an example?